globus_sdk/base.py

```diff
@@ -70,7 +70,7 @@ def __init__(
         base_path=None,
         authorizer=None,
         app_name=None,
-        http_timeout=None,
+        http_timeout: typing.Optional[float] = None,
```

Comment:
I want to get type annotations added throughout the SDK. Introducing them whenever we touch something seems like a way we can start adding them without treating it as a huge, monolithic task.